### PR TITLE
Add vkcs_db_config_group resource and datasource

### PR DESCRIPTION
**Pull request by schirevko**

### CODE REVIEW

File: vkcs/resource_vkcs_db_config_group.go [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+			},
+			"updated": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"datastore_version_id": {
```
Comment (ftersin) [inline vkcs/resource_vkcs_db_config_group.go:61]: do we really need it? do we really need it separately of datastore section?

File: vkcs/resource_vkcs_db_config_group.go [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+							Required: true,
+							ForceNew: true,
+						},
+					},
+				},
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
```
Comment (ftersin) [inline vkcs/resource_vkcs_db_config_group.go:45]: we cannot change name, but use group id to attach it to an instnace/cluster?

File: vkcs/resource_vkcs_db_config_group.go [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+	}
+
+	v := d.Get("datastore")
+	datastore, err := extractDatabaseDatastore(v.([]interface{}))
+	if err != nil {
+		return diag.Errorf("unable to determine vkcs_db_config_group datastore")
+	}
+	createOpts.Datastore = &datastore
+
+	dsParameters, err := dbDatastoreParametersGet(DatabaseV1Client, datastore.Type, datastore.Version).extract()
```
Comment (ftersin) [inline vkcs/resource_vkcs_db_config_group.go:91]: should not we put this parameter transformation to a separate func?

File: vkcs/resource_vkcs_db_config_group.go [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+
+func resourceDatabaseConfigGroup() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDatabaseConfigGroupCreate,
+		ReadContext:   resourceDatabaseConfigGroupRead,
+		UpdateContext: resourceDatabaseConfigGroupUpdate,
+		DeleteContext: resourceDatabaseConfigGroupDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
```
Comment (ftersin) [inline vkcs/resource_vkcs_db_config_group.go:19]: don't we have a way to validate schema (parameters) before call of create/update?

File: vkcs/resource_vkcs_db_config_group.go (hunk 1 above)
Comment (schirevko) [reply vkcs/resource_vkcs_db_config_group.go]: Removed

File: vkcs/db_config_group.go [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+				if valueInt, err := strconv.Atoi(value.(string)); err == nil {
+					values[name] = valueInt
+				} else {
+					return nil, fmt.Errorf("incorrect value type")
+				}
+			case "boolean":
+				if valueBool, err := strconv.ParseBool(value.(string)); err == nil {
+					values[name] = valueBool
+				} else {
+					return nil, fmt.Errorf("incorrect value type")
```
Comment (ftersin) [inline vkcs/db_config_group.go:26]: how do we let user know what parameter has incorrect type?

File: vkcs/resource_vkcs_db_config_group.go (hunk 3 above)
Comment (ftersin) [reply vkcs/resource_vkcs_db_config_group.go:89]: where?

Comment (schirevko) [reply vkcs/resource_vkcs_db_config_group.go:89]: Here: https://github.com/vk-cs/terraform-provider-vkcs/pull/25/files#diff-e53172a2e2319ef34885beb09cedc98b980a306a1efe94fdf99a36b4573d0b16R93

File: vkcs/db_config_group.go (hunk 5 above)
Comment (schirevko) [reply vkcs/db_config_group.go]: Expanded error message

File: examples/vkcs-db-config-group-create/main.tf [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    }
+    
+    configuration_id = vkcs_db_config_group.db-config-group.id
+    network {
+      uuid = vkcs_networking_network.db.id
+    }
+    flavor_id = data.vkcs_compute_flavor.db.id
+    volume_type = "ceph-ssd"
+    size = 8
+}
```
[the remark this answers is not in the document]
Comment (schirevko) [reply examples/vkcs-db-config-group-create/main.tf]: Done

File: vkcs/db_config_group.go [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+				} else {
+					return nil, fmt.Errorf("incorrect value type of parameter: %s", name)
+				}
+			case "float":
+				if valueFloat, err := strconv.ParseFloat(value.(string), 64); err == nil {
+					values[name] = valueFloat
+				} else {
+					return nil, fmt.Errorf("incorrect value type of parameter: %s", name)
+				}
+			}
```
Comment (ftersin) [inline vkcs/db_config_group.go:34]: should not we do smth in another case?

Comment (schirevko) [reply vkcs/db_config_group.go:36]: Added error message

**Review by ftersin — DISMISSED** [review]

**Review by pashapdev — APPROVED** [review]

**Pull request merged**